spec/functions/fact_for_spec.rb

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+require 'spec_helper'
+
+describe 'fact_for' do
+  let(:node_id) { '' }
+  let(:fact_name) { '' }
+  let(:fact_value) { '' }
+
+  let!(:puppetdb_query) do
+    MockFunction.new('puppetdb_query') do |f|
+      f.stubbed
+       .with(['from', 'facts', ['extract', ['value'], ['and', ['=', 'certname', node_id], ['=', 'name', fact_name]]]])
```

Comment:
This seems fragile. I also suspect that there's not really anything you can do about it.

Reply:
Yeah these are the things I tested in production. We don't have to structure the queries exactly like this, but if we do, the queries will work. I can't think of a better way, and I do think this is still better than no tests.